File: bilibili-recommend-ui/app/src/main/java/com/bilibilirecommend/base/SuperFragment.java
package com.bilibilirecommend.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * by y on 2016/9/14.
 */
public abstract class SuperFragment extends BaseFragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        if (view == null) {
            view = inflater.inflate(getLayoutId(), container, false);
        }
        initById();
        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        initData();
    }

    protected abstract void initById();

    protected abstract void initData();

    protected abstract int getLayoutId();

}

File: BilibiliRecommendUi/app/src/main/java/com/bilibilirecommend/base/SuperFragment.kt
package com.bilibilirecommend.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

abstract class SuperFragment : BaseFragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(layoutId, container, false)
    }

    protected abstract val layoutId: Int
}
